bin/live/pycbc_live_single_trigger_fits

```diff
@@ -198,110 +191,219 @@ for filename in files:
-            events[ifo] = events[ifo].select(idx_keep)
+            events[ifo] += triggers_da
+
+logging.info("All events processed")
 
 logging.info("Number of events which meet all criteria:")
 for ifo in args.ifos:
-    logging.info("{}: {} in {}s".format(ifo, len(events[ifo].data['snr']),
-                                        live_time[ifo]))
+    if ifo not in events:
+        logging.info("%s: No data", ifo)
+    else:
+        logging.info("%s: %d in %.2fs",
+                     ifo, len(events[ifo]), live_time[ifo])
 
 # split the events into bins by template duration
 logging.info('Sorting events into template duration bins')
 
 oput_file = os.path.join(args.output_directory,
-                         args.analysis_date + "-TRIGGER-FITS.hdf")
+                         args.analysis_date)
+if args.output_id_string:
+    oput_file += f'-{args.output_id_string}'
+oput_file += "-TRIGGER-FITS.hdf"
 
-fout = h5py.File(oput_file, 'w')
 # Fit the triggers within each bin
-alphas = {i: np.zeros(args.num_duration_bins, dtype=np.float32) for i in args.ifos}
-counts = {i: np.zeros(args.num_duration_bins, dtype=np.float32) for i in args.ifos}
+n_bins = duration_bin_edges.size - 1
+alphas = {i: np.zeros(n_bins, dtype=np.float32) for i in args.ifos}
+counts = {i: np.zeros(n_bins, dtype=np.float32) for i in args.ifos}
+binned_sngl_stats = {ifo: {} for ifo in args.ifos}
+
 for ifo in args.ifos:
-    for key in events[ifo].data:
-        fout['{}/triggers/{}'.format(ifo, key)] = events[ifo].data[key]
-    event_bin = np.array([tbins[d] for d in events[ifo].data['template_duration']])
-    for bin_num in range(args.num_duration_bins):
+    if ifo not in events: continue
+    # Sort the events into their bins
+    event_bin = np.array([tbins[d]
+                          for d in events[ifo].data['template_duration']])
+
+    # For each bin, do the fit
+    for bin_num in range(n_bins):
         inbin = event_bin == bin_num
+        if not np.count_nonzero(inbin):
+            # No triggers, alpha and count are -1
+            counts[ifo][bin_num] = -1
+            alphas[ifo][bin_num] = -1
+            continue
+
+        # For ease, keep the sngl_ranking values as this will be used later
+        binned_sngl_stats[ifo][bin_num] = \
+            events[ifo].data[args.sngl_ranking][inbin]
+
         counts[ifo][bin_num] = np.count_nonzero(inbin)
         alphas[ifo][bin_num], _ = trstats.fit_above_thresh(args.fit_function,
-                                      events[ifo].data[args.fit_param][inbin],
+                                      binned_sngl_stats[ifo][bin_num],
                                       args.fit_threshold)
-    fout['{}/fit_coeff'.format(ifo)] = alphas[ifo]
-    fout['{}/counts'.format(ifo)] = counts[ifo]
-    fout[ifo].attrs['live_time'] = live_time[ifo]
 
-fout['bins_upper'] = tbins.upper()
-fout['bins_lower'] = tbins.lower()
+with h5py.File(oput_file, 'w') as fout:
```

Comment:
Add a `logging.info()` just before this indicating that you are writing the results. We want to know when the fitting has finished and we are about to touch the filesystem.